lib/tpm2.c

```diff
@@ -1054,33 +1054,53 @@ tool_rc tpm2_evictcontrol(ESYS_CONTEXT *esys_context,
 }
 
 /* This function addresses ESAPI change that changes parameter type from
- * Esys_TR to TPMI_RH_HIERARCHY and breaks backwards compatibility.
+ * Esys_TR to TPMI_RH_HIERARCHY or TPMI_RH_ENABLES and breaks backwards
+ * compatibility.
  * To keep the tools parameters consistent after v4.0 release we need to
  * map the values to appropriate type based on the version of the ESYS API.
  * Note: the mapping is based on the ESYS version recognized at compile time.
  * The TSS change can be found here:
  * https://github.com/tpm2-software/tpm2-tss/pull/1531
  */
-uint32_t fix_esys_hierarchy(TPMI_RH_HIERARCHY hierarchy)
+TSS2_RC fix_esys_hierarchy(uint32_t in, uint32_t *out)
 {
 #if defined(ESYS_3_0)
-    switch (hierarchy) {
+    switch (in) {
+        case ESYS_TR_RH_NULL:
+            /* FALLTHRU */
+        case ESYS_TR_RH_OWNER:
+            /* FALLTHRU */
+        case ESYS_TR_RH_ENDORSEMENT:
+            /* FALLTHRU */
+        case ESYS_TR_RH_PLATFORM:
+            /* FALLTHRU */
+        case ESYS_TR_RH_PLATFORM_NV:
```

Comment:
I think you're right, we don't need fallthru, but they do make it explicit that you didn't intend to do something somewhere.